frame/dapps-staking/src/tests.rs

```diff
@@ -91,6 +91,57 @@ fn on_initialize_is_ok() {
     })
 }
 
+#[test]
+fn new_era_length_is_always_blocks_per_era() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+        let blocks_per_era = mock::BLOCKS_PER_ERA;
+
+        // go to beginning of an era
+        advance_to_era(mock::DappsStaking::current_era() + 1);
+
+        // record era number and block number
+        let start_era = mock::DappsStaking::current_era();
+        let starting_block_number = System::block_number();
+
+        // go to next era
+        advance_to_era(mock::DappsStaking::current_era() + 1);
+        let ending_block_number = System::block_number();
+
+        // make sure block number difference is is blocks_per_era
+        assert_eq!(mock::DappsStaking::current_era(), start_era + 1);
+        assert_eq!(ending_block_number - starting_block_number, blocks_per_era);
+    })
+}
+
+#[test]
+fn new_forced_era_length_is_always_blocks_per_era() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+        let blocks_per_era = mock::BLOCKS_PER_ERA;
+
+        // go to beginning of an era
+        advance_to_era(mock::DappsStaking::current_era() + 1);
+
+        // go to middle of era
+        run_for_blocks(1); // can be any number between 0 and blocks_per_era
+
+        // force new era
+        <ForceEra<TestRuntime>>::put(Forcing::ForceNew);
+        run_for_blocks(1); // calls on_initialize()
+
+        // note the start block number of new (forced) era
+        let start_block_number = System::block_number();
+
+        // go to start of next era
+        advance_to_era(mock::DappsStaking::current_era() + 1);
+
+        // show the length of the forced era is equal to blocks_per_era
+        let end_block_number = System::block_number();
+        assert_eq!(end_block_number - start_block_number, blocks_per_era);
+    })
+}
```

Comment:
Thanks for moving the UTs!

Please also sort out the other part of the comment.

The best way to do this is:
* prepare a new UT
* initialize first block
* enable maintenance mode
* advance to era (current_era + era_length + 3)  // or some other number
* verify that current block is higher than `NextEraStartingBlock` but that era hasn't changed
* disable maintenance mode
* advance one era
* verify era has changed and `NextEraStartingBlock` has expected value